Makefile

```diff
@@ -492,10 +492,11 @@ generate-go-conversions-kubeadm-bootstrap: $(CONVERSION_GEN) ## Generate convers
 		--go-header-file=./hack/boilerplate/boilerplate.generatego.txt \
 		./internal/apis/bootstrap/kubeadm/v1alpha3 \
 		./internal/apis/bootstrap/kubeadm/v1alpha4
-	$(MAKE) clean-generated-conversions SRC_DIRS="./bootstrap/kubeadm/types/upstreamv1beta2,./bootstrap/kubeadm/types/upstreamv1beta3,./bootstrap/kubeadm/types/upstreamv1beta4"
+	$(MAKE) clean-generated-conversions SRC_DIRS="./bootstrap/kubeadm/types/upstreamv1beta1,./bootstrap/kubeadm/types/upstreamv1beta2,./bootstrap/kubeadm/types/upstreamv1beta3,./bootstrap/kubeadm/types/upstreamv1beta4"
 	$(CONVERSION_GEN) \
 		--output-file=zz_generated.conversion.go \
 		--go-header-file=./hack/boilerplate/boilerplate.generatego.txt \
+		./bootstrap/kubeadm/types/upstreamv1beta1 \
```

Comment:
I think we don't really need conversion from upstreamv1beta1, decause we stopped generating v1beta1 kubeadm config files while ago (those types exist only because embedded in old CAPI API version).
(I will open an issue to dig into it)

Reply:
We need it. Otherwise conversion is wrong and the tests panic because of basically a stackoverflow / infinite recursion. See: https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_cluster-api/10799/pull-cluster-api-test-main/1806022375607635968 (test results of sigs.k8s.io/cluster-api/internal/apis/bootstrap/kubeadm/v1alpha3: TestFuzzyConversion/for_KubeadmConfig/spoke-hub-spoke )

This was a fun one to figure out. I can show you some details via Zoom if you want

Reply:
Basically zz_generated.conversion.go was not getting regenerated anymore
For this PR this meant that autoConvert_upstreamv1beta1_Discovery_To_v1beta1_Discovery was outdated (see the diff now on this PR). The result was that this statement
```
	out.File = (*v1beta1.FileDiscovery)(unsafe.Pointer(in.File))
```

Was assigning incompatible types, which then lead to the recursion & panic. Now the code looks like this

```go
	if in.File != nil {
		in, out := &in.File, &out.File
		*out = new(v1beta1.FileDiscovery)
		if err := Convert_upstreamv1beta1_FileDiscovery_To_v1beta1_FileDiscovery(*in, *out, s); err != nil {
			return err
		}
	} else {
		out.File = nil
	}
```

Reply:
Discussed with Fabrizio, we're good

Reply:
ack, got it
we need conversions in upstreamv1beta1 because this package is imported in v1beta1, and conversion.go in v1beta1 to expects to import conversion func from upstreamv1beta1